README.md

```diff
@@ -1,3 +1,148 @@
 # StellarWP Changelog Embed
 
-A simple block for embedding a changelog text file as content.
+A WordPress plugin that allows you to display changelogs from your GitHub repositories in a clean, organized format using a Gutenberg block.
+
+## Features
+
+- Connects to GitHub to fetch changelog.txt files directly from your repositories
+- Parses WordPress plugin-style changelog entries
+- Displays changelogs in an expandable/collapsible interface
+- Groups changes by type (Fix, Feature, Tweak, Security, etc.)
+- Caches GitHub API responses to improve performance
+- Full Gutenberg block integration with live preview
+- Responsive design that works on all devices
+- Preserves user preferences for expanded/collapsed versions
+
+## Installation
+
+1. Upload the `stellar-changelog-embed` folder to the `/wp-content/plugins/` directory
+2. Activate the plugin through the 'Plugins' menu in WordPress
+3. Add a StellarWP Changelog Embed block to any post or page
+
+## Usage
+
+### Using the Block
+
+1. Add a "StellarWP Changelog Embed" block to your post or page
+2. In the block settings, enter:
+   - Repository Owner (GitHub username or organization)
+   - Repository Name
+   - Changelog File Path (defaults to "changelog.txt")
+   - Branch (defaults to "main")
+   - Maximum Versions to display
+3. Save your post or page
+
+### GitHub API Configuration
+
+To avoid GitHub API rate limits or to access private repositories:
+
+1. Go to Settings > StellarWP Changelog Embed
+2. Enter your GitHub Personal Access Token
+3. Save Changes
+
+## Changelog Format
+
+The plugin expects changelog files in this format:
+
+```
+= [4.21.1] =
```

Comment:
We started to use something like ` = [4.3.3] 2025-08-06 =`

Reply:
I updated the example changelog as based on the implementation in stellarwp/changelogger, I believe that's the format everyone should be moving to if they aren't already there. It will support basic `= [x.x.x] =` as well though :)